lib/core/src/animation/physics.rs

```diff
@@ -0,0 +1,308 @@
+use crate::prelude::*;
+
+use crate::animation::{FixedStepAnimator, ContinuousTimeAnimator, linear_interpolation};
+
+use nalgebra::Vector3;
+use nalgebra::zero;
+use crate::traits::HasPosition;
+
+
+
+// ====================
+// === PhysicsForce ===
+// ====================
+
+pub trait PhysicsForce {
+    fn force(&self, kinematics:&KinematicsProperties) -> Vector3<f32>;
+}
+
+// ======================
+// === DragProperties ===
+// ======================
+
+/// This structure contains air dragging properties.
+#[derive(Default, Clone, Copy)]
+pub struct DragProperties {
+    coefficient: f32
+}
+
+impl DragProperties {
+    pub fn new(amount:f32) -> Self {
+        Self { coefficient: amount }
+    }
+}
+
+impl PhysicsForce for DragProperties {
+    fn force(&self, kinematics:&KinematicsProperties) -> Vector3<f32> {
+        let velocity  = kinematics.velocity;
+        let speed     = velocity.norm();
+        if speed > 0.0 {
+            velocity.normalize() * speed.powf(2.0) * -0.5 * self.coefficient
```

Comment:
What is this `* -0.5` ?  I believer it should not be in this equation. Moreover, I may be wrong, but `speed.powf(2.0)` is not the fastest way of computing a power of 2.

Reply:
https://github.com/rust-lang/rust/issues/47234

Reply:
@wdanilo the simplified formula is `F = 1/2 * coefficient * velocity ^ 2`. This is the intensity of the drag force. But the actual drag force is opposite to the velocity, thus I am negating it.

I will replace `speed.powf(2.0)` by `speed * speed`.